sdk/tests/invariant.test.ts

```diff
@@ -248,4 +269,240 @@ describe('invariant', async () => {
     const pools = await invariant.getPools(account)
     assert.deepEqual(pools.length, 1)
   })
+  describe('positions', async () => {
+    const lowerTickIndex = -20n
+    const upperTickIndex = 10n
+
+    beforeEach(async () => {
+      if (!token0.contract?.address || !token1.contract?.address || !invariant.contract?.address) {
+        throw new Error()
+      }
+
+      const feeTier = newFeeTier({ v: 6000000000n }, 10)
+      await invariant.addFeeTier(account, feeTier)
+
+      await invariant.createPool(
+        account,
+        token0.contract.address.toString(),
+        token1.contract.address.toString(),
+        feeTier,
+        { v: 1000000000000000000000000n },
+        0n
+      )
+
+      await token0.approve(account, invariant.contract.address.toString(), 10000000000n)
+      await token1.approve(account, invariant.contract.address.toString(), 10000000000n)
+
+      const poolKey = convertObj(
+        newPoolKey(token0.contract.address.toString(), token1.contract.address.toString(), feeTier)
+      )
+
+      const pool = await invariant.getPool(
+        account,
+        token0.contract.address.toString(),
+        token1.contract.address.toString(),
+        feeTier
+      )
+
+      await invariant.createPosition(
+        account,
+        poolKey,
+        lowerTickIndex,
+        upperTickIndex,
+        { v: 1000000000000n },
+        pool.sqrtPrice,
+        pool.sqrtPrice
+      )
+      const position = await invariant.getPosition(account, account.address, 0n)
+      const expectedPosition: Position = {
+        poolKey: poolKey,
+        liquidity: { v: 1000000000000n },
+        lowerTickIndex: lowerTickIndex,
+        upperTickIndex: upperTickIndex,
+        feeGrowthInsideX: { v: 0n },
+        feeGrowthInsideY: { v: 0n },
+        lastBlockNumber: 0n,
+        tokensOwedX: 0n,
+        tokensOwedY: 0n
+      }
+      await positionEquals(position, expectedPosition)
+    })
+    it('remove position', async () => {
```

Comment:
There is a lack of 'create position' entry. I'm guessing it is fully inside 'before'. You can transfer the getPosition and assert inside 'create position'. There is no need to repeat the same assertion inside all tests